### ISSUE TITLE
v2: Camera parameter does not seem to have an effect on depth estimation

### ISSUE THREAD

**Issue by erikfastermann** [lpiccinelli-eth/UniDepth#105]
Hi,

we are really happy with your work and how well your model performs in our test.

However we have one problem: The optional camera parameter does not seem to have an effect on the depth result in our test (for our quick and dirty Python test code see the attachment below).

Inside `unidepth/models/unidepthv2/decoder.py` in ` run_camera` the `where_use_gt_rays` looks like it is always zero, so the `rays_gt` camera info is not used.

While skimming the paper, it sounds like the camera parameters are not required, but can still be used (also shown in the example in the README).

Could you please clarify if this is intentional or what we might have misunderstood?

Thank you for your help!

[unidepth_test.txt](https://github.com/user-attachments/files/19246274/unidepth_test.txt)

**Comment by lpiccinelli-eth** [comment]
Thank you for your appreciation and question!
Indeed you are right, the model can use GT camera (and rays) if provided.
However, it works only when the model is in eval mode, as eval model set `self.training` to `False` and it avoids that part of the code within the `if self.training` under `run_camera`.  Anyway, I am removing that if clause to limit future misunderstandings (#104 ) .
